src/test/java/seedu/address/model/person/PriorityTest.java

```diff
@@ -0,0 +1,119 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class PriorityTest {
+    private static final Priority PRIORITY_HIGH = new Priority("high");
+    private static final Priority PRIORITY_MEDIUM = new Priority("medium");
+    private static final Priority PRIORITY_LOW = new Priority("low");
+
+    private static final Priority PRIORITY_HIGH_CAPS = new Priority("HIGH");
+    private static final Priority PRIORITY_MEDIUM_CAPS = new Priority("MEDIUM");
+    private static final Priority PRIORITY_LOW_CAPS = new Priority("LOW");
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Priority(null));
+    }
+
+    @Test
+    public void constructor_invalidType_throwsIllegalArgumentException() {
+        String invalidPriority = "";
+        assertThrows(IllegalArgumentException.class, () -> new Priority(invalidPriority));
+    }
+
+    @Test
+    public void isValidPriority() {
+        // null type
+        assertThrows(NullPointerException.class, () -> Priority.isValidPriority(null));
+
+        // invalid type
+        assertFalse(Priority.isValidPriority("")); // empty string
+        assertFalse(Priority.isValidPriority(" ")); // spaces only
+        assertFalse(Priority.isValidPriority("^")); // only non-alphanumeric characters
+        assertFalse(Priority.isValidPriority("peter*")); // contains non-alphanumeric characters
+
+        assertFalse(Priority.isValidPriority("12345")); // numbers only
+        assertFalse(Priority.isValidPriority("peter the 2nd")); // alphanumeric characters
+        assertFalse(Priority.isValidPriority("Capital Tan")); // with capital letters
+
+        assertTrue(Priority.isValidPriority("high")); //high is valid
+        assertTrue(Priority.isValidPriority("medium")); //medium is valid
+        assertTrue(Priority.isValidPriority("low")); //low is valid
+        assertTrue(Priority.isValidPriority("HIGH")); //case insensitive
+        assertTrue(Priority.isValidPriority("MEDIUM")); //case insensitive
+        assertTrue(Priority.isValidPriority("LOW")); //case insensitive
```

Comment:
Should use the constant that you declared??

Reply:
Rectified